File: javascript/medium/sieve.js
export const primes = (limit) => {
  let possiblePrimes = [];
  for (let i = 2; i <= limit; i++) {
    possiblePrimes.push(i);
  }
  for (let i = 2; i <= limit; i++) {
    for (let j = 2 * i; j <= limit; j += i) {
      if (possiblePrimes.indexOf(j) !== -1) {
        console.log(possiblePrimes.indexOf(j));
        possiblePrimes.splice(possiblePrimes.indexOf(j), 1);
      }
    }
  }
  return possiblePrimes;
};

File: python/easy/sieve.py
def primes(limit):
    composite = set()
    all_numbers = set(range(2, limit + 1))
    for i in all_numbers:
        for j in range(2, limit + 1):
            composite.add(i * j)
    primes = sorted(list(all_numbers - composite))
    return primes